src/vs/workbench/contrib/terminalContrib/chatAgentTools/test/browser/bufferOutputPolling.test.ts

```diff
@@ -18,6 +20,15 @@ suite('racePollingOrPrompt', () => {
 	const defaultLanguageModelsService = {} as any;
 	const defaultExecution = { getOutput: () => 'output' };
 
+	test('getOutput enforces 16000 character limit', () => {
+		const longString = 'A'.repeat(17000);
+		const terminal = new RawXtermTerminal();
+		terminal.write(longString, () => {
+			const output = getOutput(terminal);
+			assert.strictEqual(output, longString.slice(-16000));
+		});
+	});
```

Comment:
Since it's a callback, you need an `async () =>` test. Use this helper for write:

```ts
	function write(data: string): Promise<void> {
		return new Promise<void>((resolve) => {
			xterm.write(data, resolve);
		});
	}
```